### PR TITLE
Refactor utils to test sc processor

### PR DESCRIPTION
**Pull request by stefanandy**
refactored utils to testScProcessor

### CODE REVIEW

File: process/smartContract/testScProcessor.go [hunk 1]
```diff
@@ -9,16 +9,23 @@ import (
...
+	*scProcessor
+}
+
+func NewTestScProcessor(internalData *scProcessor) *TestScProcessor {
+	return &TestScProcessor{internalData}
+}
+
 // GetLatestTestError should only be used in tests!
 // It locates the latest error in the collection of smart contracts results
 // TODO remove this file as it is a horrible hack to test some conditions
```
Comment (iulianpascalau) [inline process/smartContract/testScProcessor.go:25]: remove TODO

File: process/smartContract/testScProcessor.go [hunk 2]
```diff
@@ -9,16 +9,23 @@ import (
...
 )
 
+// TestScProcessor extends scProcessor and is used in tests as it exposes some functions
+// that are not supposed to be used in production code
+// Exported functions simplify the reproduction of edge cases
+type TestScProcessor struct {
+	*scProcessor
+}
+
+func NewTestScProcessor(internalData *scProcessor) *TestScProcessor {
```
Comment (iulianpascalau) [inline process/smartContract/testScProcessor.go:19]: Add comment

File: process/smartContract/testScProcessor.go [hunk 3]
```diff
@@ -9,16 +9,23 @@ import (
...
+	return &TestScProcessor{internalData}
+}
+
 // GetLatestTestError should only be used in tests!
```
Comment (iulianpascalau) [inline process/smartContract/testScProcessor.go:23]: refactor comment

File: integrationTests/vm/txsFee/utils/utils.go [hunk 4]
```diff
@@ -180,13 +179,13 @@ func ProcessSCRResult(
 	expectedErr error,
 ) {
 	scProcessor := testContext.ScProcessor
-	scrProcessor, ok := scProcessor.(process.SmartContractResultProcessor)
-	require.True(t, ok)
+	//require.True(t, ok)
```
Comment (iulianpascalau) [inline integrationTests/vm/txsFee/utils/utils.go:182]: remove comment

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

**Review by Oxsy-SIbiu — APPROVED** [review]

**Review by SebastianMarian — DISMISSED** [review]

File: process/smartContract/testScProcessor.go [hunk 5]
```diff
@@ -16,13 +16,12 @@ type TestScProcessor struct {
...
 
+// NewTestScProcessor -
 func NewTestScProcessor(internalData *scProcessor) *TestScProcessor {
 	return &TestScProcessor{internalData}
 }
 
-// GetLatestTestError should only be used in tests!
-// It locates the latest error in the collection of smart contracts results
-// TODO remove this file as it is a horrible hack to test some conditions
+// GetLatestTestError It locates the latest error in the collection of smart contracts results
```
Comment (iulianpascalau) [inline process/smartContract/testScProcessor.go:24]: Remove "It"

**Review by iulianpascalau — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by valentincraciun99 — APPROVED** [review]
